src/psyclone/tests/psyir/nodes/call_test.py

```diff
@@ -628,6 +629,539 @@ def test_call_get_callees_local(fortran_reader):
     assert result == [psyir.walk(Routine)[1]]
 
 
+def test_call_get_callee_1_simple_match(fortran_reader):
+    '''
+    Check that right routine has been found.
+    '''
+    code = '''
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f, g
+    call foo(e, f, g)
+  end subroutine
+
+  ! Matching routine
+  subroutine foo(a, b, c)
+    integer :: a, b, c
+  end subroutine
+
+end module some_mod'''
+
+    psyir = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = psyir.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+
+    result = call_foo.get_callee()
+
+    routine_match: Routine = psyir.walk(Routine)[1]
+    assert result is routine_match
+
+
+def test_call_get_callee_2_optional_args(fortran_reader):
+    '''
+    Check that optional arguments have been correlated correctly.
+    '''
+    code = '''
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f, g
+    call foo(e, f)
+  end subroutine
+
+  ! Matching routine
+  subroutine foo(a, b, c)
+    integer :: a, b
+    integer, optional :: c
+  end subroutine
+
+end module some_mod'''
+
+    root_node: Node = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = root_node.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    routine_match: Routine = root_node.walk(Routine)[1]
+    assert routine_match.name == "foo"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+    assert call_foo.routine.name == "foo"
+
+    arg_idx_list = []
+    result: Routine = call_foo.get_callee(ret_arg_match_list=arg_idx_list)
```

Comment:
As I say in the comments earlier, it's more usual for us to return a tuple in situations where we want more than one quantity back from a routine. So this would be `result, arg_list = call_foo.get_callee(...)`

Reply:
OK.